Dockerfile.dcos-e2e

```diff
@@ -0,0 +1,23 @@
+FROM python:3-alpine
+
+ARG DCOS_E2E_VERSION=2018.12.10.0
+
+RUN apk add --no-cache --virtual .build \
+        git \
+        gcc \
+        libc-dev \
+        libffi-dev \
+        libressl-dev \
+        make \
+ && apk add --no-cache \
+        bash \
+        jq \
+        docker \
+        openssh-client \
+        sed \
+ && git -c advice.detachedHead=false \
+        clone --quiet --depth 1 --branch "${DCOS_E2E_VERSION}" \
+        https://github.com/dcos/dcos-e2e.git /usr/src/dcos-e2e \
+ && pip3 install /usr/src/dcos-e2e \
```

Comment:
I'm not aware of an advantage of this over `pip3 install git+https://github.com/dcos/dcos-e2e.git@${DCOS_E2E_VERSION}`

Reply:
git+https urls clone the entire history, during development it was noticeable that it was taking a while to build so I switched it to this. I guess installing from the 'archive' url like in requirements-dev.txt would do the same thing.

Reply:
Good to know - another reason to switch to PyPI which I hope I can do sometime.